src/main.js

```diff
@@ -1,144 +1,175 @@
+// delay, we are *not* requiring at the top of this file. Use idleCallbacks to
+// attempt to pre-cache requires for fast resolution if scheduler allows. Since
+// all requires are statically defined and inside some function, Atom should be
+// able "snapshot" the package.
+//
+const dependencies = [
+  () => require('./validate/editor'),
+  () => require('./linter-message'),
+  () => require('./worker-manager'),
+  () => require('./eslint-config-inspector'),
+  () => require('./debug'),
+  () => require('./rules'),
+  () => require('path'),
+  () => require('./atom-config'),
+]
+
+// Miscellaneous idle tasks.
+//
+const idleTasks = [
+  // Pre-start the worker
+  () => require('./worker-manager').task.start(),
+  // Install peer packages
+  () => require('atom-package-deps').install('linter-eslint'),
+]
+
+// Run idle tasks with all reasonable attempts to avoid blocking.
+//
+const makeIdleRunner = () => {
+  const { requestIdleCallback, cancelIdleCallback } = window
+  let id
+  let tasks
+
+  const loadLazily = (deadline) => {
+    // While time alotted by scheduler and work remaining
+    while (deadline.timeRemaining() && tasks.length) {
+      // Remove and run function from end of list.
+      tasks.pop()()
+    }
 
-const makeIdleCallback = (work) => {
-  let callbackId
-  const callBack = () => {
-    idleCallbacks.delete(callbackId)
-    work()
+    // If work remains, request more time
+    if (tasks.length) id = requestIdleCallback(loadLazily)
   }
-  callbackId = window.requestIdleCallback(callBack)
-  idleCallbacks.add(callbackId)
-}
 
-const scheduleIdleTasks = () => {
-  const linterEslintInstallPeerPackages = () => {
-    require('atom-package-deps').install('linter-eslint')
-  }
-  const linterEslintLoadDependencies = loadDeps
-  const linterEslintStartWorker = () => {
-    loadDeps()
-    worker.task.start()
+  return (requestedTasks) => {
+    // Spread avoids external mutation and allows non-array iterable
+    tasks = [...requestedTasks]
+    // Init first request
+    id = requestIdleCallback(loadLazily)
+    // Return a disposable
+    return { dispose: () => cancelIdleCallback(id) }
   }
+}
 
-  if (!atom.inSpecMode()) {
-    makeIdleCallback(linterEslintInstallPeerPackages)
-    makeIdleCallback(linterEslintLoadDependencies)
-    makeIdleCallback(linterEslintStartWorker)
+// // Simplified composite disposable avoids requiring atom core in main
+// //
+const makeCompositeDisposable = () => {
+  const disposables = new Set()
+  return {
+    add: (disposable) => {
+      if (!(disposable.dispose instanceof Function)) {
+        throw new Error('disposable must have dispose method')
+      }
+      disposables.add(disposable)
+    },
+    dispose: () => disposables.forEach(d => d.dispose())
   }
 }
 
+
 module.exports = {
   activate() {
-    getMigrations().map(makeIdleCallback)
-    this.subscriptions = new CompositeDisposable()
+    // const { CompositeDisposable } = require('atom')
```

Comment:
🔥 If we _are_ switching, we don't need commented out code here...

Reply:
Yep. artifact from my A/B testing.